storage-proofs/porep/src/stacked/vanilla/create_label/multi.rs

```diff
@@ -205,17 +211,21 @@ fn create_layer_labels(
     exp_labels: Option<&mut MmapMut>,
     num_nodes: u64,
     cur_layer: u32,
+    core_group: Arc<Option<MutexGuard<Vec<CoreIndex>>>>,
 ) -> Result<()> {
     info!("Creating labels for layer {}", cur_layer);
     // num_producers is the number of producer threads
-    let (lookahead, num_producers, producer_stride) = {
-        // NOTE: Stride must not exceed `sdr_parents_cache_window_nodes`.
-        // If it does, the process will deadlock with producers and consumers
-        // waiting for each other.
-        // TODO: Enforce this.
-        //(800, 1, 128)
-        (800, 2, 128)
-    };
+    let settings = settings::SETTINGS.lock().expect("settings lock failure");
+    let lookahead = settings.multicore_sdr_lookahead;
+    let num_producers = settings.multicore_sdr_producers;
+
+    // NOTE: Stride must not exceed `sdr_parents_cache_window_nodes`.
+    // If it does, the process will deadlock with producers and consumers
+    // waiting for each other.
+    let cache_window_nodes = parents_cache.size() / (4 * DEGREE);
+    let producer_stride = settings
+        .multicore_sdr_producer_stride
+        .min(cache_window_nodes as u64);
 
     const BYTES_PER_NODE: usize = (NODE_SIZE * DEGREE) + 64;
```

Comment:
Comment to describe what this `64` and the above `4` are representing?

Reply:
I converted all the magic numbers to self-describing constants.